yamux/src/session.rs

```diff
@@ -618,24 +618,26 @@ where
             );
         }
 
-        while let Some(ref mut interval) = self.keepalive {
+        if let Some(ref mut interval) = self.keepalive {
             match Pin::new(interval).as_mut().poll_next(cx) {
                 Poll::Ready(Some(_)) => {
                     self.keep_alive(cx, Instant::now())?;
                 }
                 Poll::Ready(None) => {
                     debug!("yamux::Session poll keepalive interval finished");
-                    break;
                 }
-                Poll::Pending => break,
+                Poll::Pending => (),
             }
         }
-
-        loop {
+        let mut need_wake = false;
+        for _ in 0..16 {
             if self.is_dead() {
                 break;
```

Comment:
`need_wake` can be true or false here. Should we assign a value here?

Reply:
break here means yamux session is dead, the entire structure will be dropped, no need to care about whether it needs to be executed next time